docs/spec/governance/state.md

```diff
@@ -2,47 +2,58 @@
 
 ## State
 
-### Procedures
+### Procedures and base types
 
 `Procedures` define the rule according to which votes are run. There can only 
 be one active procedure at any given time. If governance wants to change a 
 procedure, either to modify a value or add/remove a parameter, a new procedure 
 has to be created and the previous one rendered inactive.
 
+
 ```go
+type Procedure struct {
+  VotingPeriod      int64               //  Length of the voting period. Initial value: 2 weeks
+  MinDeposit        sdk.Coins           //  Minimum deposit for a proposal to enter voting period. 
+  Threshold         rational.Rational   //  Minimum propotion of Yes votes for proposal to pass. Initial value: 0.5
+  Veto              rational.Rational   //  Minimum value of Veto votes to Total votes ratio for proposal to be vetoed. Initial value: 1/3
+  MaxDepositPeriod  int64               //  Maximum period for Atom holders to deposit on a proposal. Initial value: 2 months
+  GovernancePenalty sdk.Rat               //  Penalty if validator does not vote
+  
+  IsActive          bool                //  If true, procedure is active. Only one procedure can have isActive true.
+}
+```
 
-type VoteType byte
+The current active procedure is stored in a global `params` KVStore.
+
+And some basic types:
+
+```go
+
+type Vote byte
 
 const (
-    VoteTypeYes         = 0x1
-    VoteTypeNo          = 0x2
-    VoteTypeNoWithVeto  = 0x3
-    VoteTypeAbstain     = 0x4
+    VoteYes         = 0x1
+    VoteNo          = 0x2
+    VoteNoWithVeto  = 0x3
+    VoteAbstain     = 0x4
 )
 
 type ProposalType  byte
 
 const (
-    ProposalTypePlainText = 0x1
+    ProposalTypePlainText       = 0x1
     ProposalTypeSoftwareUpgrade = 0x2
```

Comment:
Will have ParameterChangeProposal for MVP

Reply:
I'm not sure this is a good idea. This has major code implications (would require new tx types, files to store changeable parameters, pull parameters from file, and more if we want to activate/deactivate modules). Can't we launch with SoftwareUpgrade only and implement Paramter Proposals just after?

Reply:
What do you mean by files?  It's just a normal keeper/store called `globalParams`

Reply:
Does this keeper already exist?